plugins/destination/mongodb/client/read.go

```diff
@@ -50,15 +52,31 @@ func (c *Client) reverseTransform(f arrow.Field, bldr array.Builder, val any) er
 	case *array.BinaryBuilder:
 		b.Append(val.(primitive.Binary).Data)
 	case *array.TimestampBuilder:
-		b.Append(arrow.Timestamp((val).(primitive.DateTime).Time().UTC().UnixMicro()))
+		switch b.Type().(*arrow.TimestampType).Unit {
```

Comment:
Can also do:
```go
ts, err := arrow.TimestampFromString(t.Format(time.RFC3339Nano), b.Type().(*arrow.TimestampType).Unit)
```

Reply:
I think we can ask to provide a helper func in the upstream Arrow code for this (time.Time + arrow.TimeUnit -> arrow.Timestamp)